modules/imgproc/src/warp_kernels.simd.hpp

```diff
@@ -2130,6 +1592,1414 @@ void warpAffineLinearApproxInvoker_8UC4(const uint8_t *src_data, size_t src_step
+        CV_INSTRUMENT_REGION();
+
+        const auto *src = src_data;
+        auto *dst = dst_data;
+        size_t srcstep = src_step/sizeof(float), dststep = dst_step/sizeof(float);
+        int srccols = src_cols, srcrows = src_rows;
+        int dstcols = dst_cols;
+        float M[9];
+        for (int i = 0; i < 9; i++) {
+            M[i] = static_cast<float>(dM[i]);
+        }
+        float bval[] = {
+            saturate_cast<float>(border_value[0]),
+            saturate_cast<float>(border_value[1]),
+            saturate_cast<float>(border_value[2]),
+            saturate_cast<float>(border_value[3]),
+        };
+        int border_type_x = border_type != BORDER_CONSTANT &&
+                            border_type != BORDER_TRANSPARENT &&
+                            srccols <= 1 ? BORDER_REPLICATE : border_type;
+        int border_type_y = border_type != BORDER_CONSTANT &&
+                            border_type != BORDER_TRANSPARENT &&
+                            srcrows <= 1 ? BORDER_REPLICATE : border_type;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+        constexpr int max_vlanes_32{VTraits<v_float32>::max_nlanes};
+        constexpr int max_uf{max_vlanes_32*2};
+        int vlanes_32 = VTraits<v_float32>::vlanes();
+        // unrolling_factor = lane_size / 16 = vlanes_32 * 32 / 16 = vlanes_32 * 2
+        int uf = vlanes_32 * 2;
+
+        std::array<float, max_vlanes_32> start_indices;
+        std::iota(start_indices.data(), start_indices.data() + max_vlanes_32, 0.f);
+
+        v_uint32 inner_srows = vx_setall_u32((unsigned)srcrows - 2),
+                 inner_scols = vx_setall_u32((unsigned)srccols - 1),
+                 outer_srows = vx_setall_u32((unsigned)srcrows + 1),
+                 outer_scols = vx_setall_u32((unsigned)srccols + 1);
+        v_float32 delta = vx_setall_f32(static_cast<float>(uf));
+        v_int32 one = vx_setall_s32(1), three = vx_setall_s32(3);
+        v_int32 v_srcstep = vx_setall_s32(int(srcstep));
+        int32_t addr[max_uf],
+                src_ix[max_uf],
+                src_iy[max_uf];
+        float pixbuf[max_uf*4*3];
+
+        float bvalbuf[max_uf*3];
+        for (int i = 0; i < uf; i++) {
+            bvalbuf[i*3] = bval[0];
+            bvalbuf[i*3+1] = bval[1];
+            bvalbuf[i*3+2] = bval[2];
+        }
+        v_float32 bval_v0_l = vx_load(&bvalbuf[0]);
+        v_float32 bval_v0_h = vx_load(&bvalbuf[vlanes_32]);
+        v_float32 bval_v1_l = vx_load(&bvalbuf[uf]);
+        v_float32 bval_v1_h = vx_load(&bvalbuf[uf+vlanes_32]);
+        v_float32 bval_v2_l = vx_load(&bvalbuf[uf*2]);
+        v_float32 bval_v2_h = vx_load(&bvalbuf[uf*2+vlanes_32]);
+#endif
+
+        for (int y = r.start; y < r.end; y++) {
+            float* dstptr = dst + y*dststep;
+            int x = 0;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+            CV_WARPPERSPECTIVE_LINEAR_VECTOR_COMPUTE_MAPPED_COORD1();
+
+            for (; x < dstcols - uf; x += uf) {
```

Comment:
the correct condition should be `x <= dstcols - uf`, I believe

Reply:
Yes, thanks. Corrected.